tests/unit/core/api/test_user.js

```diff
@@ -0,0 +1,37 @@
+/* global window */
+import config from 'config';
+
+import * as api from 'core/api/user';
+import { CLIENT_APP_ANDROID } from 'core/constants';
+import {
+  createApiResponse,
+  createUserProfileResponse,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let mockWindow;
+  const apiHost = config.get('apiHost');
+
+  beforeEach(() => {
+    mockWindow = sinon.mock(window);
+  });
+
+  describe('userProfile', () => {
+    const mockResponse = () => createApiResponse({
+      jsonData: createUserProfileResponse(),
+    });
+
+    it('fetches the current user profile', () => {
+      mockWindow.expects('fetch')
+        .withArgs(`${apiHost}/api/v3/accounts/profile/?lang=en-US`)
+        .once()
+        .returns(mockResponse());
+
+      return api.userProfile({
+        api: { clientApp: CLIENT_APP_ANDROID, lang: 'en-US' },
```

Comment:
This should use `dispatchClientMetadata().store.getState().api`. I know it's way more verbose/work but we should definitely get away from stubbing out raw state. We used to do it and it always leads to refactors or oddness.